File: faucet.js
const axios = require('axios');
const FormData = require('form-data');

// 2Captcha API key
const API_KEY = 'YOUR_2CAPTCHA_API_KEY';

// Wallet address yang akan digunakan untuk klaim faucet
const walletAddress = 'YOUR_WALLET_ADDRESS';

// URL faucet
const faucetUrl = 'https://0g-faucet.corenodehq.xyz'; 

// Site key hCaptcha dari halaman faucet
const siteKey = 'dcaaabbe-8882-43b0-b102-c5d69e54a91e'; 

// Fungsi untuk mendapatkan hCaptcha response dari 2Captcha
async function getCaptchaSolution() {
  try {
    // Kirim permintaan untuk menyelesaikan hCaptcha ke 2Captcha
    const response = await axios.post('http://2captcha.com/in.php', null, {
      params: {
        key: API_KEY,
        method: 'hcaptcha',
        sitekey: siteKey,
        pageurl: faucetUrl,
      },
    });

    const captchaId = response.data.split('|')[1];
    console.log('Captcha ID:', captchaId);

    // Tunggu beberapa detik sampai 2Captcha menyelesaikan captcha
    let result;
    while (true) {
      result = await axios.get('http://2captcha.com/res.php', {
        params: {
          key: API_KEY,
          action: 'get',
          id: captchaId,
        },
      });

      if (result.data.includes('OK|')) {
        return result.data.split('|')[1]; // Token hCaptcha
      }

      // Tunggu beberapa detik sebelum mencoba lagi
      await new Promise(resolve => setTimeout(resolve, 5000));
    }
  } catch (error) {
    console.error('Error menyelesaikan captcha:', error);
    throw error;
  }
}

// Fungsi untuk mengklaim faucet
async function claimFaucet() {
  try {
    // Mendapatkan token hCaptcha
    const hcaptchaResponse = await getCaptchaSolution();
    console.log('Token hCaptcha:', hcaptchaResponse);

    // Kirim klaim faucet ke situs
    const form = new FormData();
    form.append('address', walletAddress);
    form.append('h-captcha-response', hcaptchaResponse);

    const response = await axios.post(faucetUrl + '/submit', form, {
      headers: form.getHeaders(),
    });

    if (response.status === 200) {
      console.log('Faucet klaim berhasil!');
    } else {
      console.log('Error klaim faucet:', response.status);
    }
  } catch (error) {
    console.error('Error dalam klaim faucet:', error);
  }
}

// Menjalankan klaim faucet
claimFaucet();

File: faucet.py
import time
import requests
from twocaptcha import TwoCaptcha
from dotenv import load_dotenv
import os

# Load variabel dari file .env
load_dotenv()

# Ambil API key dan wallet addresses dari file .env
API_KEY_2CAPTCHA = os.getenv('API_KEY_2CAPTCHA')
WALLET_ADDRESSES = os.getenv('WALLET_ADDRESSES').splitlines()  # Memisahkan wallet address yang berada di setiap baris

# Site key hCaptcha yang sudah ditemukan
site_key = 'dcaaabbe-8882-43b0-b102-c5d69e54a91e'
url = 'https://0g-faucet.corenodehq.xyz'  # URL faucet

# Inisialisasi solver 2Captcha
solver = TwoCaptcha(API_KEY_2CAPTCHA)

def solve_hcaptcha(wallet_address):
    try:
        # Meminta 2Captcha untuk menyelesaikan hCaptcha
        result = solver.hcaptcha(sitekey=site_key, url=url)

        # Mendapatkan token dari hasil
        hcaptcha_token = result['code']
        print(f"Token hCaptcha untuk wallet {wallet_address}: {hcaptcha_token}")

        # Kirim token ke server untuk klaim faucet
        response = requests.post(
            url='https://0g-faucet.corenodehq.xyz/submit',
            data={
                'address': wallet_address,  # Menggunakan wallet address dari daftar
                'h-captcha-response': hcaptcha_token,
            }
        )
        
        if response.status_code == 200:
            print(f"Faucet claim berhasil untuk wallet {wallet_address}!")
        else:
            print(f"Error saat klaim faucet untuk wallet {wallet_address}: {response.status_code}")
    
    except Exception as e:
        print(f"Terjadi kesalahan saat klaim faucet untuk wallet {wallet_address}: {e}")

def claim_for_all_wallets():
    # Menjalankan klaim untuk semua wallet yang ada di daftar WALLET_ADDRESSES
    for wallet_address in WALLET_ADDRESSES:
        solve_hcaptcha(wallet_address.strip())  # Menghapus spasi yang tidak perlu

if __name__ == '__main__':
    claim_for_all_wallets()
